### DISCUSSION TITLE
[WASM] Understanding CI/CD and UI testing

### DISCUSSION THREAD

**Discussion by DierkDroth** [unoplatform/uno#8710] [Q&A] [answered]
I'm trying to get my head around on what's involved on getting UNO CI/CD/UI testing up and running - primarily for WASM. I'm aware of [this ](https://platform.uno/blog/an-introduction-to-unit-testing-for-uno-platform-applications/)and [this](https://github.com/unoplatform/Uno.UITest) but still would have a couple of (dummy?) questions:

- is there any reference on how to setup CI/CD/UI testing for UNO WASM on GitHub? 
- is GitHub the right platform for that in the first place? If not, what else should I look into?
- running UI tests definitely requires a 'true session' .. which likely would be some (virtual?) server, no? Or is there any other option with a smaller footprint?

Like I said: I'm a dummy and might have more questions after I had an idea on the basics above...

Thanks in advance

**Comment by jeromelaban** [comment, marked as answer]
> * is there any reference on how to setup CI/CD/UI testing for UNO WASM on GitHub?

We do not have such documentation in uno directly, but https://platform.uno/blog/how-to-uno-platform-and-azure-pipelines is still valid.

> * is GitHub the right platform for that in the first place? If not, what else should I look into?
GitHub or Azure devops are appropriate.

> * running UI tests definitely requires a 'true session' .. which likely would be some (virtual?) server, no? Or is there any other option with a smaller footprint?
Most platforms don't need an interactive session, except windows, but Uno does not provide UI testing on Windows for WPF or Skia targets yet. You can follow https://github.com/unoplatform/Uno.UITest to get updates on when we'll add support for Skia and/or Windows targets.

**Reply by DierkDroth** [reply]
Thanks @jeromelaban. That gives me something to work with...
